DigitalLearningSolutions.Data.Tests/DataServices/LearningLogItemsDataServiceTests.cs

```diff
@@ -147,91 +151,158 @@ public void GetLearningLogItems_returns_all_learning_hub_resource_log_items_for_
                     firstActivityName,
-                    firstResourceLink,
-                    competencyLearningResourceId
+                    firstResourceLink
                 );
-                service.InsertLearningLogItem(
+                AssertLearningLogItemHasCorrectValuesForLearningHubResource(
+                    result[1],
                     delegateId,
                     addedDate,
+                    competencyLearningResourceId,
                     secondActivityName,
-                    secondResourceLink,
-                    competencyLearningResourceId
-                );
-                service.InsertLearningLogItem(
-                    differentDelegateId,
-                    addedDate,
-                    "activity 3",
-                    "resource link 3",
-                    competencyLearningResourceId
+                    secondResourceLink
                 );
+            }
+        }
 
-                // When
-                var result = service.GetLearningLogItems(delegateId).ToList();
+        [Test]
+        public void GetLearningLogItem_returns_null_for_non_learning_hub_resources()
+        {
+            // When
+            var result = service.GetLearningLogItem(2);
 
-                // Then
-                using (new AssertionScope())
-                {
-                    result.Count.Should().Be(2);
-                    AssertLearningLogItemHasCorrectValuesForLearningHubResource(
-                        result[0],
-                        delegateId,
-                        addedDate,
-                        competencyLearningResourceId,
-                        firstActivityName,
-                        firstResourceLink
-                    );
-                    AssertLearningLogItemHasCorrectValuesForLearningHubResource(
-                        result[1],
-                        delegateId,
-                        addedDate,
-                        competencyLearningResourceId,
-                        secondActivityName,
-                        secondResourceLink
-                    );
-                }
-            }
-            finally
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void GetLearningLogItem_returns_learning_log_item()
+        {
+            // Given
+            var addedDate = new DateTime(2021, 11, 1);
+
+            using var transaction = new TransactionScope();
+            var itemId = InsertLearningLogItem(
+                GenericDelegateId,
+                addedDate,
+                GenericCompetencyLearningResourceId,
+                GenericActivityName,
+                GenericResourceLink
+            );
+
+            // When
+            var result = service.GetLearningLogItem(itemId);
+
+            // Then
+            using (new AssertionScope())
             {
-                transaction.Dispose();
+                result.Should().NotBeNull();

```

Comment:
Do we still need the var transaction if we're removing transaction.Dispose() ?

Reply:
Good call, replaced the declaration and just wrapped it all in a standard using block.

Reply:
I generally don't have a problem with `var transaction` - it means you don't need the braces or nesting which I find quite nice in tests. But either is fine.

Reply:
Ah didn't know that! It does look nice without the braces, but I guess we won't worry too much about it